lib/EnsEMBL/REST/Controller/Sequence.pm

```diff
@@ -188,39 +198,60 @@ sub _process_feature {
   my $slice;
   my $desc;
   my $mask_feature = $c->request->param('mask_feature');
-  
+
   #Translations
   if($object->isa('Bio::EnsEMBL::Translation')) {
     $molecule = 'protein';
     $seq = $object->transcript()->translate()->seq();
+    # Grab the subsequence if we've been asked to trim the ends
+    $seq = $self->_process_subseq($c, $seq) if($c->stash()->{dosubseq});
   }
   #Transcripts
   elsif($object->isa('Bio::EnsEMBL::PredictionTranscript')) {
     if($type eq 'cdna') {
       $seq = $object->spliced_seq($mask_feature);
+      # Grab the subsequence if we've been asked to trim the ends
+      $seq = $self->_process_subseq($c, $seq) if($c->stash()->{dosubseq});
```

Comment:
Abundant copy-paste implies there might be room for simplification here, with only a single _process_subseq() call for any of the $seq outputs. Clarity should take priority if the subseq can be easily overlooked.